File: CodingSamples/Java/Language-Runtime/AbstractTypeTest/bank1/src/banking/CurrentAccount.java
package banking;

//a class declared with 'final' modifier cannot be used
//as a super-class
final class CurrentAccount extends Account {

    public void withdraw(double amount) throws InsufficientFundsException {
        balance -= amount;
    }

    public void deposit(double amount) {
        if(balance < 0)
            amount *= 0.9;
        balance += amount;
    }
}

File: CodingSamples/DotNet/Language-Runtime/AbstractTypeTest/BankLib2/CurrentAccount.cs
namespace Banking;

//a class declared with 'sealed' modifier cannot be used
//as a base class
sealed class CurrentAccount : Account, IFineable
{
    public override void Deposit(double amount)
    {
        Balance += amount;
    }

    public override void Withdraw(double amount)
    {
        Balance -= amount;
    }

    //explicit interface implementation
    bool IFineable.Withdraw(double penalty)
    {
        if(Balance < 0)
        {
            Balance -= penalty;
            return true;
        }
        return false;
    }
}